packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -2483,6 +2483,18 @@ moduleFor('Components test: curly components', class extends RenderingTest {
     this.assertText('Jackson');
   }
 
+  ['@test injecting an unknown service raises an exception']() {
+    this.registerComponent('foo-bar', {
+      ComponentClass: Component.extend({
+        missingService: inject.service()
+      })
+    });
+
+    throws(() => {
```

Comment:
I'm going to wrap the `validateInjections` in a `runInDebug` and change the `Error` to an assert